frontend/layouts/discover-page/helpers.ts

```diff
@@ -47,3 +58,123 @@ export const useSortingByOptions = (target?: SortingByTargetType): SortingOption
     ...(sortingOptions[target] || []),
   ]) as SortingOptionType[];
 };
+
+const pathFilters = {
+  [Paths.Projects]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[only_verified]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.ProjectDevelopers]: [
+    'filter[category]',
+    'filter[impact]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.Investors]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[full_text]',
+  ],
+  [Paths.OpenCalls]: [
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[only_verified]',
+    'filter[full_text]',
+  ],
+};
+
+const getPathParams = (
+  path: string,
+  pathname: string,
+  queryParams: { [key: string]: string | string[] }
+) => {
+  const { page, sorting, search, ...filters } = queryParams;
+
+  // Get only applicable filters for each path to avoid unnecessary fetches
+  const pathFilter = pickBy(filters, (_, key) => pathFilters[path].includes(key));
+
+  const params: { [key: string]: string | string[] } = { page, search, ...pathFilter };
+
+  if (pathname !== path) {
+    return {
+      ...params,
+      fields: ['name'],
+      perPage: 1,
+      'page[size]': 1,
+    };
+  }
+
+  // Use sorting value only when applies to path to avoid unnecessary fetches
+  if (path === Paths.Projects || sorting?.includes('name') || sorting?.includes('created_at')) {
+    params.sorting = sorting;
+  }
+
+  if (pathname === Paths.Projects) {
+    return {
+      ...params,
+      includes: ['project_developer', 'involved_project_developers'],
+      'fields[project_developer]': 'picture,name,contact_phone,contact_email',
+      'fields[involved_project_developers]': 'picture,name,contact_phone,contact_email',
+    };
+  }
+  if (pathname === Paths.Investors || pathname === Paths.ProjectDevelopers) {
+    return { ...params, perPage: 9 };
+  }
+  if (pathname === Paths.OpenCalls) {
+    return { ...params, includes: ['investor'] };
```

Comment:
Could we limit the amount of data we load from the investor as you did for the PD and involved PD when loading projects?